dockerfiles/base/scripts/base/library.sh

```diff
@@ -345,9 +344,17 @@ verify_version_compatibility() {
   ##      - If they don't match and one is nightly, fail
   ##      - If they don't match, then if CLI is older fail with message to get proper CLI
   ##      - If they don't match, then if CLLI is newer fail with message to run upgrade first
-
   CHE_IMAGE_VERSION=$(get_image_version)
 
+  # Only check for newer versions if not in offline mode.
+  if ! is_offline; then
+    NEWER=$(compare_versions $CHE_IMAGE_VERSION)
```

Comment:
When I started developing this feature I was only going to do the check if it was latest.  But while implementing it, I saw more value in it for the tagged versions!  

The tagged versions was how I was able to implement some tests for it.  But I thought about leaving it in and then discussing the trade offs in this issue with others.